Capri.Services/Promoters/PromoterCreator.cs

```diff
@@ -13,32 +14,41 @@ public class PromoterCreator : IPromoterCreator
         private readonly ISqlDbContext _context;
         private readonly IMapper _mapper;
         private readonly IUserCreator _userCreator;
+        private readonly IInstituteGetter _instituteGetter;
 
         public PromoterCreator(
             ISqlDbContext context,
+            IMapper mapper,
             IUserCreator userCreator,
-            IMapper mapper)
+            IInstituteGetter instituteGetter
+            )
         {
             _context = context;
-            _userCreator = userCreator;
             _mapper = mapper;
+            _userCreator = userCreator;
+            _instituteGetter = instituteGetter;
         }
 
         public async Task<IServiceResult<PromoterViewModel>> Create(
             PromoterRegistration registration)
         {
+            var instituteResult = await _instituteGetter.Get(registration.InstituteId);
+            if(!instituteResult.Successful())
```

Comment:
Same here. Where is it used next?

Reply:
The same as above. We have to check if the institute exists